From 64744cd74473fea26985204b05877b7d352a6d53 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Sun, 5 May 2024 23:14:12 +0200
Subject: [PATCH 0834/1695] core: rng_support: helper function to register TRNG

Add helper function for hw_get_random_bytes() drivers implementation
to register when their TRNG is ready to  operate.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 62a4e139137b519e20d1366f1fc40b0946ac7ecb core: rng_support: helper function to register TRNG]
Change-Id: Iee78e51177f28b44f7f11f7ba09c2d90f6872059
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/377165
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/include/rng_support.h | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/core/include/rng_support.h b/core/include/rng_support.h
index 1dc4d891a..f18eb9227 100644
--- a/core/include/rng_support.h
+++ b/core/include/rng_support.h
@@ -1,12 +1,19 @@
 /* SPDX-License-Identifier: BSD-2-Clause */
 /*
- * Copyright (c) 2014, STMicroelectronics International N.V.
+ * Copyright (c) 2014-2024, STMicroelectronics International N.V.
  */
 #ifndef __RNG_SUPPORT_H__
 #define __RNG_SUPPORT_H__
 
+#include <crypto/crypto.h>
 #include <stdint.h>
 
+/* Get random bytes from a hardware entropy source */
 TEE_Result hw_get_random_bytes(void *buf, size_t blen);
 
+/* Register RNG support, especially to the HWRNG PTA */
+static inline void hw_register_get_random_bytes(void)
+{
+	crypto_rng_register_hw();
+}
 #endif /* __RNG_SUPPORT_H__ */
-- 
2.47.2

